### PR TITLE
Fix deprecated warning caused by shuffle_and_repeat when using make_csv_dataset

### PR DESCRIPTION
**Pull request by yongtang**
This fix fixes unnecessary deprecated warning caused by shuffle_and_repeat when using make_csv_dataset:
```
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow_core/python/data/experimental/ops/readers.py:215: shuffle_and_repeat (from tensorflow.python.data.experimental.ops.shuffle_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.Dataset.shuffle(buffer_size, seed)` followed by `tf.data.Dataset.repeat(count)`. Static tf.data optimizations will take care of using the fused implementation.
```

Note the fix doesn't use shuffle().repeat() as was suggested in the warning message, instead this fix keeps the original implementation to avoid potential performance issues.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/data/experimental/ops/readers.py [hunk 1]
```diff
@@ -209,9 +209,10 @@ def _maybe_shuffle_and_repeat(
   """Optionally shuffle and repeat dataset, as requested."""
   if num_epochs != 1 and shuffle:
     # Use shuffle_and_repeat for perf
-    return dataset.apply(
-        shuffle_ops.shuffle_and_repeat(shuffle_buffer_size, num_epochs,
-                                       shuffle_seed))
+    def _apply_fn(input_dataset):  # pylint: disable=missing-docstring
```
Comment (jsimsa) [inline tensorflow/python/data/experimental/ops/readers.py]: you can just return `shuffle_ops._ShuffleAndRepeatDataset(...)` instead of doing the `apply`

Comment (jsimsa) [reply tensorflow/python/data/experimental/ops/readers.py]: and for consistency, it might make sense to do the same for `parallel_interleave` (instead of the `_parallel_interleave` indirection you introduced)

**Review by jsimsa — DISMISSED** [review]

File: tensorflow/python/data/experimental/ops/interleave_ops.py [hunk 2]
```diff
@@ -105,8 +88,12 @@ def parallel_interleave(map_func,
     A `Dataset` transformation function, which can be passed to
     `tf.data.Dataset.apply`.
   """
-  return _parallel_interleave(map_func, cycle_length, block_length, sloppy,
-                              buffer_output_elements, prefetch_input_elements)
+  def _apply_fn(dataset):
+    return readers.ParallelInterleaveDataset(
```
Comment (jsimsa) [inline tensorflow/python/data/experimental/ops/interleave_ops.py]: you can just return readers.ParallelInterleaveDataset here

**Comment by yongtang** [comment]
Thanks @jsimsa for the review. There was a test failure (passed a typo name). I have updated the PR and pushed. Can you look and approve again if everything is still fine? Sorry for the extra inconvenience.

**Review by jsimsa — APPROVED** [review]
